### PR TITLE
ICDS Dashboard: Bar chart responsiveness

### PR DESCRIPTION
**Pull request by saiprakash-v**
Trello links: 
https://trello.com/c/UnAWxi84
https://trello.com/c/WIBuC7St
https://trello.com/c/0lN5ACOz

##### SUMMARY
removed hardcoded width for charts, enabling chart to take width based on container.
removed the custom tooltip and enabled the default one (to solve tooltip going out of view).
split labels which has more than 12 characters into smaller chunks of size 12 each to enable wrapping (could not break the string using css when text overflows in "svg text")

### CODE REVIEW

File: custom/icds_reports/static/icds_reports/mobile/css/map-chart.less
Comment (czue) [inline custom/icds_reports/static/icds_reports/mobile/css/map-chart.less]: what behavior does this change? do you have like a before/after?

File: custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js
Comment (czue) [reply custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js]: https://caniuse.com/#feat=arrow-functions only possible issue is IE11. but maybe easier to not use this

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/map-or-sector-view.directive.html
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/map-or-sector-view.directive.html]: I think you can also delete the CSS classes associated with this.

File: custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js]: does this have any negative UI implications on web? curious why this limit was here in the first place.

**Review by czue — CHANGES_REQUESTED** [review]
can you respond to comments? also, can you confirm you've tested extensively with actual data on both web and mobile?

File: custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js]: Ok. I will replace this with a for loop

File: custom/icds_reports/static/icds_reports/mobile/css/map-chart.less
Comment (saiprakash-v) [reply custom/icds_reports/static/icds_reports/mobile/css/map-chart.less]: Before: when labels get closer, after a certain limit labels are hidden to avoid overlapping
Screenshot from 2020-01-31 18-00-30

after:
Screenshot from 2020-01-31 18-01-49

File: custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js]: Keeping fixed width makes page horizontally scrollable on smaller views like this. 
Screenshot from 2020-01-31 18-14-45
Now it would take the container width.

**Comment by saiprakash-v** [comment]
> can you respond to comments? also, can you confirm you've tested extensively with actual data on both web and mobile?

Yes, I have tested with actual data

File: custom/icds_reports/static/icds_reports/mobile/css/map-chart.less
Comment (czue) [reply custom/icds_reports/static/icds_reports/mobile/css/map-chart.less]: :+1:

File: custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js
Comment (czue) [reply custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js]: gotcha. that seems like an improvement to me. trying to think of any reason it's not always better (maybe something with dense charts on small screens?). or maybe it's just a relic no one ever bothered thinking about till now.

**Comment by czue** [comment]
@saiprakash-v tests are failing on this
```
1 failing
  1) AWC Reports Directive tests chart options:
     AssertionError: expected undefined to equal 1100
      at http://localhost:8000/static/chai/chai.js:210
      at http://localhost:8000/static/chai/chai.js:2178
      at http://localhost:8000/static/icds_reports/js/spec/awc-reports.directive.spec.js:114
```

File: custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js
Comment (czue) [inline custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js]: Why is this reversed now?

**Review by czue — APPROVED** [review]
good to go once tests pass

**Comment by czue** [comment]
added "don't merge" until @saiprakash-v looks at an issue with "see more"

**Review by czue — APPROVED** [review]

**Pull request merged**
